### PR TITLE
Add RVF support

### PR DESCRIPTION
**Pull request by adrianf0**
This MR request adds RVF support (#33).

### CODE REVIEW

File: src/avtp_rvf.c [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+#define MASK_EVT (BITMASK(4) << SHIFT_EVT)
+#define MASK_PD (BITMASK(1) << SHIFT_PD)
+#define MASK_I (BITMASK(1) << SHIFT_I)
+#define MASK_RAW_PIXEL_DEPTH (BITMASK(4) << SHIFT_RAW_PIXEL_DEPTH)
+#define MASK_RAW_PIXEL_FORMAT (BITMASK(4) << SHIFT_RAW_PIXEL_FORMAT)
+#define MASK_RAW_FRAME_RATE (BITMASK(8) << SHIFT_RAW_FRAME_RATE)
+#define MASK_RAW_COLORSPACE (BITMASK(4) << SHIFT_RAW_COLORSPACE)
+#define MASK_RAW_NUM_LINES (BITMASK(4) << SHIFT_RAW_NUM_LINES)
+#define MASK_RAW_I_SEQ_NUM (BITMASK(8) << SHIFT_RAW_I_SEQ_NUM)
+#define MASK_RAW_LINE_NUMBER (BITMASK(16) << SHIFT_RAW_LINE_NUMBER)
```
[the remark this answers is not in the document]
Comment (edersondisouza) [reply src/avtp_rvf.c:67]: Hmmm... but the kernel is full of examples of tab alignment, such as https://github.com/torvalds/linux/blob/master/kernel/sched/sched.h#L454 or https://github.com/torvalds/linux/blob/master/include/linux/netdevice.h#L102.

File: include/avtp_rvf.h [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+/* RVF 'pixel format' field values. */
+#define AVTP_RVF_PIXEL_FORMAT_MONO 0x00
+#define AVTP_RVF_PIXEL_FORMAT_411 0x01
+#define AVTP_RVF_PIXEL_FORMAT_420 0x02
+#define AVTP_RVF_PIXEL_FORMAT_422 0x03
+#define AVTP_RVF_PIXEL_FORMAT_444 0x04
+#define AVTP_RVF_PIXEL_FORMAT_4224 0x06
+#define AVTP_RVF_PIXEL_FORMAT_4444 0x07
+#define AVTP_RVF_PIXEL_FORMAT_BAER_GRBG 0x08
```
Comment (edersondisouza) [inline include/avtp_rvf.h:54]: Small typo here `BAYER`, not `BAER`.

**Review by edersondisouza — COMMENTED** [review]
Looking great! Just a typo to fix.
The tab thing is your call, unless someone else complains.

File: include/avtp_rvf.h (hunk 2 above)
Comment (adrianf0) [reply include/avtp_rvf.h]: Fixed. Thanks!

File: src/avtp_rvf.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianf0) [reply src/avtp_rvf.c:67]: No strings attached to this tab formatting on my side. Simply clang checks formatting automatically using the .[clang-format](https://github.com/Avnu/libavtp/blob/3cf00b36bbe3e50e3f0beb445c84735bd10289d9/.clang-format) file I have added in this pull request. IMHO, in order to be consistent, all sources should be aligned with the formatting defined in this file at a certain point.

File: include/avtp_rvf.h [hunk 3]
```diff
@@ -51,10 +51,10 @@ extern "C" {
...
 #define AVTP_RVF_PIXEL_FORMAT_4444 0x07
-#define AVTP_RVF_PIXEL_FORMAT_BAER_GRBG 0x08
-#define AVTP_RVF_PIXEL_FORMAT_BAER_RGGB 0x09
-#define AVTP_RVF_PIXEL_FORMAT_BAER_BGGR 0x0A
-#define AVTP_RVF_PIXEL_FORMAT_BAER_GBRG 0x0B
+#define AVTP_RVF_PIXEL_FORMAT_BAYER_GRBG 0x08
+#define AVTP_RVF_PIXEL_FORMAT_BAYER_RGGB 0x09
+#define AVTP_RVF_PIXEL_FORMAT_BAYER_BGGR 0x0A
+#define AVTP_RVF_PIXEL_FORMAT_BAYER_GBRG 0x0B
 #define AVTP_RVF_PIXEL_FORMAT_USER 0x0F
```
Comment (edersondisouza) [inline include/avtp_rvf.h:58]: Cool, but please squash this into the "Add RVF support" patch.

Comment (adrianf0) [reply include/avtp_rvf.h:58]: Done.

File: src/avtp_rvf.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianf0) [reply src/avtp_rvf.c]: Fixed. I modified the orignal Linux [.clang-foramt](https://github.com/Avnu/libavtp/blob/cc4bd01edd8ce98b22a7fe947d827071e80b8104/.clang-format) to align the consecutive macros.

**Comment by edersondisouza** [comment]
lgtm!

**Review by vcgomes — APPROVED** [review]

**Comment by adrianf0** [comment]
@edersondisouza : I have added support to Gstreamer ([link](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/1335)). Feel free to take a look.

**Comment by edersondisouza** [comment]
> @edersondisouza : I have added support to Gstreamer ([link](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/1335)). Feel free to take a look.

Great! I'll take a look in the next few days =D

**Pull request merged**